agora/footfall_ai/src/video_capture.py

```diff
@@ -0,0 +1,38 @@
+import cv2
+import time
+import threading
+import queue
+import cv2, queue, threading, time
+
+# bufferless VideoCapture
+class VideoCapture:
+    def __init__(self, name):
+        self.cap = cv2.VideoCapture(name)
+        self.q = queue.Queue()
```

Comment:
We should keep a limit on the queue size by providing maxsize.
https://stackoverflow.com/questions/39678374/python-queue-queue-best-practice-to-set-size

The producer in this case is faster compared to the consumer and will lead to our queue growing very large and can lead to running out of memory.

Reply:
@niachary - Agree could you please push your suggested changes?

Reply:
@fcabrera23 Do you want me to push the changes to the same PR branch?

Reply:
@niachary yes - Feel free to push there

Reply:
I dont have the permission to push but this is what I would do:

self.q = queue.Queue(**maxsize=150**)

**while self.q.full():
                time.sleep(0.01)**
                if not self.running:
                    return
            
            self.q.put(cv2.resize(frame, (1280, 720)))

Reply:
@niachary just added you as part of the team

Reply:
Pushed new changes:
https://github.com/Azure/jumpstart-apps/pull/19/commits/5bb74287582ca14b79d79edb88a37a6e905ba8c3